src/Compilers/VisualBasic/Portable/Compilation/MethodCompiler.vb

```diff
@@ -1180,7 +1179,9 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
                 Dim compilation = compilationState.Compilation
                 If compilation.ShouldAddEvent(method) Then
                     If block Is Nothing Then
-                        compilation.SymbolDeclaredEvent(sourceMethod)
+                        If Not method.IsPartial OrElse method.PartialImplementationPart Is Nothing Then
```

Comment:
Can we even get here with the partial definition method? I think we always switch to implementation before calling CompileMethod. If this is correct, I would prefer that we add an assert to that fact at the beginning of the method and revert this change.